release/ray_release/file_manager/job_file_manager.py

```diff
@@ -149,6 +166,26 @@ def upload(self, source: Optional[str] = None, target: Optional[str] = None):
 
     def delete(self, target: str, recursive: bool = False):
         def delete_fn():
+            if self.cloud_storage_provider == "s3":
+                delete_aws_fn()
+                return
+            if self.cloud_storage_provider == "gs":
+                delete_gs_fn()
+                return
+
+        def delete_gs_fn():
+            if recursive:
+                blobs = self.gs_client.list_blobs(
+                    self.bucket,
+                    prefix=target,
+                )
+                for blob in blobs:
+                    blob.delete()
+            else:
+                blob = self.gs_client.bucket(self.bucket).blob(target)
+                blob.delete()
+
+        def delete_aws_fn():
```

Comment:
flatten these out rather than using a closure?

with a closure, it is hard to tell what vars are used and what are not, and can be error prone as the value of the vars can be changed (sometimes even after the code that references them).